File: Data Structures & Algorithms/trapping-rain-water/submission-1.cpp
class Solution {
public:
    int trap(vector<int>& height) {
        int l=0,r=height.size()-1,ml=height[0],mr=height[height.size()-1], ans=0;

        while(r>l){
            if(ml<=mr){
                l++;
                ml = max(ml,height[l]);
                ans+= ml - height[l];
            }
            else{
                r--;
                mr = max(mr,height[r]);
                ans+=mr-height[r];
            }
        }

        return ans;
    }
};

File: Data Structures & Algorithms/balanced-binary-tree/submission-1.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    boolean val = true;
    public boolean isBalanced(TreeNode root) {
        if(root==null){
            return true;
        }

        int l = traverse(root.left);
        int r = traverse(root.right);

        if(Math.abs(l-r)>1){
            val=false;
        }

        return val;
    }

    public int traverse(TreeNode root){
        if(root==null){
            return 0;
        }

        int l = traverse(root.left);
        int r = traverse(root.right);

        if(Math.abs(l-r)>1){
            val=false;
        }
        return Math.max(l,r)+1;
    }
}
